### PR TITLE
MGDAPI-4655 apply priorityClassName to all rhoam crs

### PR DESCRIPTION
**Pull request by cecobask**
# Issue link
https://issues.redhat.com/browse/MGDAPI-4655

# What
The priority field for the rhmi-operator pod is currently set to 0 (the lowest possible value). This means that if there weren't enough free resources on a cluster, the rhmi-operator pod would be one of the first things to be scaled down in order to free up resources.

Audited the relevant pods created by RHOAM and standardized their priority values where needed.

# Verification steps
- Provision an OSD cluster
- Prepare the cluster and install RHOAM from the latest stable version `v1.29.0`:
```
git checkout tags/rhoam-v1.29.0 -b rhoam-v1.29.0

make cluster/prepare/local LOCAL=false

cat << EOF | oc create -f -
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: rhoam-operators
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/tdimov0/managed-api-service-index:1.29.0
EOF

make deploy/integreatly-rhmi-cr.yml LOCAL=false
```
- Install the RHOAM operator from Operator Hub
- Wait until the RHOAM installation completes successfully
- Patch the catalog source to point at `1.30.0` and upgrade RHOAM:
```
oc patch catalogsource rhoam-operators -n openshift-marketplace --type=json -p='[{"op": "replace", "path": /spec/image, "value": "quay.io/tdimov0/managed-api-service-index:1.30.0"}]'
```
- Wait until the upgrade completes and verify that the resources below have a `priorityClassName` field set to the corresponding value in column **PriorityClassName**.  

**NOTE**: Italic **PriorityClassName** value means the resource was standardized as part of this task. Non-italic value means the resource already had a **PriorityClassName** set.  
The entries with missing priority can be on-cluster postgres or redis instances. This scenario can happen when a cluster is ran with `useClusterStorage: true`. Thread link: https://chat.google.com/room/AAAAssKWhTI/Bj2d-wNKOVY

## Deployments
 **Namespace**                             | **Name**                                  | **PriorityClassName**                 
-------------------------------------------|-------------------------------------------|---------------------------------------
 redhat-rhoam-3scale                       | marin3r-instance                          | *rhoam-pod-priority*                  
 redhat-rhoam-3scale-operator              | threescale-operator-controller-manager-v2 | *rhoam-pod-priority*                  
 redhat-rhoam-cloud-resources-operator     | cloud-resource-operator                   | *rhoam-pod-priority*                  
 redhat-rhoam-customer-monitoring-operator | grafana-operator-controller-manager       | *rhoam-pod-priority*                  
 redhat-rhoam-customer-monitoring-operator | grafana-deployment                        | rhoam-pod-priority                    
 redhat-rhoam-marin3r                      | ratelimit                                 | rhoam-pod-priority                    
 redhat-rhoam-marin3r-operator             | marin3r-controller-webhook                | *rhoam-pod-priority*                  
 redhat-rhoam-marin3r-operator             | marin3r-controller-manager                | *rhoam-pod-priority*                  
 redhat-rhoam-observability                | prometheus-operator                       | observability-operator-priority-class 
 redhat-rhoam-observability                | grafana-operator                          | observability-operator-priority-class 
 redhat-rhoam-observability                | grafana-deployment                        | observability-operator-priority-class 
 redhat-rhoam-observability-operator       | observability-operator-controller-manager | observability-operator-priority-class 
 redhat-rhoam-operator                     | threescale-redis-rhoam                    | -                                     
 redhat-rhoam-operator                     | threescale-postgres-rhoam                 | -                                     
 redhat-rhoam-operator                     | threescale-backend-redis-rhoam            | -                                     
 redhat-rhoam-operator                     | rhssouser-postgres-rhoam                  | -                  
 redhat-rhoam-operator                     | rhsso-postgres-rhoam                      | -                                     
 redhat-rhoam-operator                     | rhmi-operator                             | rhoam-pod-priority                    
 redhat-rhoam-operator                     | ratelimit-service-redis-rhoam             | -                                     
 redhat-rhoam-rhsso-operator               | rhsso-operator                            | *rhoam-pod-priority*                  
 redhat-rhoam-user-sso-operator            | rhsso-operator                            | *rhoam-pod-priority*                  

## DeploymentConfigs
 **Namespace**       | **Name**           | **PriorityClassName** 
---------------------|--------------------|-----------------------
 redhat-rhoam-3scale | zync-que           | rhoam-pod-priority    
 redhat-rhoam-3scale | zync-database      | rhoam-pod-priority    
 redhat-rhoam-3scale | zync               | rhoam-pod-priority    
 redhat-rhoam-3scale | system-sphinx      | rhoam-pod-priority    
 redhat-rhoam-3scale | system-sidekiq     | rhoam-pod-priority    
 redhat-rhoam-3scale | system-memcache    | rhoam-pod-priority    
 redhat-rhoam-3scale | system-app         | rhoam-pod-priority    
 redhat-rhoam-3scale | backend-worker     | rhoam-pod-priority    
 redhat-rhoam-3scale | backend-listener   | rhoam-pod-priority    
 redhat-rhoam-3scale | backend-cron       | rhoam-pod-priority    
 redhat-rhoam-3scale | apicast-staging    | rhoam-pod-priority    
 redhat-rhoam-3scale | apicast-production | rhoam-pod-priority    

## StatefulSets
 **Namespace**              | **Name**                  | **PriorityClassName**                 
----------------------------|---------------------------|---------------------------------------
 redhat-rhoam-observability | prometheus-prometheus     | observability-operator-priority-class 
 redhat-rhoam-observability | alertmanager-alertmanager | observability-operator-priority-class 
 redhat-rhoam-rhsso         | keycloak                  | rhoam-pod-priority                    
 redhat-rhoam-user-sso      | keycloak                  | rhoam-pod-priority

### CODE REVIEW

**Review by KevFan — COMMENTED** [review]
Looks good 👍 

Have one comment 🧑‍💻

File: pkg/products/cloudresources/reconciler.go [hunk 1]
```diff
@@ -203,6 +203,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
...
+		ctx,
+		client,
+		fmt.Sprintf("cloud-resources.v%s", integreatlyv1alpha1.OperatorVersionCloudResources),
+		r.Config.GetOperatorNamespace(),
+		r.installation.Spec.PriorityClassName,
+	)
+	if err != nil || phase == integreatlyv1alpha1.PhaseFailed {
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile cloud-resources csv deployments priority class name", err)
+		return phase, err
+	}
```
[the remark this answers is not in the document]
Comment (cecobask) [reply pkg/products/cloudresources/reconciler.go:216]: Yes, it was intended to not block the reconcile 👍🏻 It's using the `PatchIfExists` function behind the hood

**Comment by KevFan** [comment]
Verified pods have `priorityClassName` field set on upgrade 👍 

/lgtm

**Comment by austincunningham** [comment]
@cecobask might need a rebase to fix the code coverage

**Pull request merged**